samtranslator/model/appsync.py

```diff
@@ -7,9 +7,23 @@
 from samtranslator.model.types import IS_DICT, IS_STR, is_type, list_of
 from samtranslator.utils.types import Intrinsicable
 
+# Data source constants can be found here under "Type" property:
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-datasource.html
+SUPPORTED_DATASOURCES = {"AMAZON_DYNAMODB"}
 
-class Auth(TypedDict, total=False):
-    Type: str
+
+class DeltaSyncConfigType(TypedDict):
```

Comment:
This applies to any new classes/functions/variables. We should avoid putting stuff under anywhere else than `internal` if possible. People depend on the library in strangest of ways, so that's one obvious place that conveys we don't provide any guarantees.

Reply:
Offline discussion: Moving the `appsync` module to internal is trivial, so I have updated as such. Moving the `sam_resources` macros are not, the translator searches this module for all macros when resolving which makes separation difficult. I will look later to see if we can add multiple modules.